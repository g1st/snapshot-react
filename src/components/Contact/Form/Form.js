import React, { Component } from 'react';

import Button from '../../Button';
import { Wrapper, FormContainer, Row, Name, Email, Message } from './styled';

class Form extends Component {
  state = {
    message: '',
    name: '',
    email: '',
    width: window.innerWidth
  };

  componentDidMount() {
    window.addEventListener('resize', this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  // don't rerender on width change
  shouldComponentUpdate(nextProps, nextState) {
    return (
      nextState.message !== this.state.message ||
      nextState.name !== this.state.name ||
      nextState.email !== this.state.email
    );
  }

  handleResize = e => {
    const windowWidth = e.target.innerWidth;

    this.setState(() => ({
      width: windowWidth
    }));
  };

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    alert(`Comment was sent: ${this.state.message}`);
  }

  render() {
    return (
      <Wrapper>
        <FormContainer onSubmit={e => this.handleSubmit(e)}>
          <Row>
            <Name
              required
              type="text"
              name="name"
              placeholder="Name"
              value={this.state.name}
              onChange={e => this.handleChange(e)}
            />
            <Email
              required
              type="email"
              name="email"
              placeholder="Email"
              value={this.state.email}
              onChange={e => this.handleChange(e)}
            />
          </Row>
          <Message
            name="message"
            value={this.state.message}
            placeholder="Message"
            rows="8"
            onChange={e => this.handleChange(e)}
            required
          />
          <Button
            type="submit"
            big={this.state.width < 768}
            title="Send message"
          />
        </FormContainer>
      </Wrapper>
    );
  }
}

export default Form;
