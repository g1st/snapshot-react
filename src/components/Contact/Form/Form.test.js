import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Form from '.';

describe('Form Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(<Form />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
