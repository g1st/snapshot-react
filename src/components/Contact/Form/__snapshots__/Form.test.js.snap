// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component matches snapshot 1`] = `
<Wrapper>
  <FormContainer
    onSubmit={[Function]}
  >
    <Row>
      <Name
        name="name"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="text"
        value=""
      />
      <Email
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </Row>
    <Message
      name="message"
      onChange={[Function]}
      placeholder="Message"
      required={true}
      rows="8"
      value=""
    />
    <Button
      big={false}
      title="Send message"
      type="submit"
    />
  </FormContainer>
</Wrapper>
`;
